src/main/java/software/amazon/msk/auth/iam/internals/AuthenticationRequestParams.java

```diff
@@ -57,9 +57,14 @@ public String getServiceScope() {
 
     public static AuthenticationRequestParams create(@NonNull String host,
             AWSCredentials credentials,
-            @NonNull String userAgent) throws IllegalArgumentException {
```

Comment:
An alternate fix proposal: It does not require user to explicitly pass the region. 
The original intention is that the region is derived from the MSK hostname. 

But the root cause for failure is is due to `RegionMetadataFactory.create()` (at line 41 above) returning a Legacy Metadata object due to some runtime classpath file. The legacy metadata one can not derive region from hostname.

How about directly creating RegionMetadata in MSK library as a fix? 
i.e. Create regular region metadata at the line linked above to something like `new RegionMetadata(new PartitionsLoader().build())`.

Reply:
Hello @rangadi I would prefer this approach because adding a new config will be a one door decision for us and we need to maintain it for future releases as well.

Reply:
@plazma-prizma @rangadi I have proposed the alternate pull request here: https://github.com/aws/aws-msk-iam-auth/pull/134

Let me know what you think!

Reply:
Thanks @nihalpot. #134 looks great.